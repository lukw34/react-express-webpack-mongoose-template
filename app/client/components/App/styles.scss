$imagSize: 150px;
$imageCounter: 2;
$margin: 75px;
.title {
    display: flex;
    margin: auto;
    flex-direction: column;
    justify-content: center;
    margin-top: 50px;
    .titleText {
        background-color: $primary;
        border: solid $secondary-light;
        border-radius: 20px;
        font-size: 1.5rem;
        text-align: center;
        padding-top: 20px;
    }
    .imageContainer {
        display: flex;
        flex-direction: row;
        position: relative;
        justify-content: center;
        @for $i from 1 through $imageCounter {
            .image-#{$i} {
                position: absolute;
                width: $imagSize;
                left: ($i - 1) * $margin; 
                right: 0;
                height: $imagSize;
            }
        }
    }
}