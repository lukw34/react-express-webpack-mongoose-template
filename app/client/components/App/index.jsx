import React from 'react';

import styles from './styles.scss';
import webpack from '../../shared/img/webpack_logo.png';
import react from '../../shared/img/react_logo.png';

const App = () => (
    <div className={styles.title}>
        <div className={styles.titleText}>react-expres-webpack-mongoose-template</div>
        <div className={styles.imageContainer}>
            {
                [react, webpack].map((src, i) => (
                    <img
                        className={styles[`image-${i + 1}`]}
                        key={src}
                        src={src}
                        alt={src}
                    />))
            }
        </div>
    </div>
);

export default App;