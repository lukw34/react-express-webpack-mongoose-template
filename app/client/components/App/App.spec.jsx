import React from 'react';
import {shallow} from 'enzyme';

import App from './index';

describe('Example <App /> test.', () => {
    it(' render component.', () => {
        const component = shallow(<App technologies={['test']}/>);
        expect(component).toMatchSnapshot();
    });
});