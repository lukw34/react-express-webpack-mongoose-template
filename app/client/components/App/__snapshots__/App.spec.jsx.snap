// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example <App /> test.  render component. 1`] = `
<div
  className="App"
>
  <Technology
    key="test"
    name="test"
  />
</div>
`;
